File: Python3/1267. Count Servers that Communicate.py
class Solution:
    def countServers(self, grid: List[List[int]]) -> int:
        m, n = len(grid), len(grid[0])

        rows = [0] * m
        cols = [0] * n

        connected = 0

        for r in range(m):
            for c in range(n):
                if grid[r][c] == 1:
                    rows[r] += 1
                    cols[c] += 1
                    

        for r in range(m):
            for c in range(n)            :
                if grid[r][c]:
                    if rows[r] > 1 or cols[c] > 1:
                        connected += 1

        return connected

                    

File: Java/1267. Count Servers that Communicate.java
class Solution {
    public int countServers(int[][] grid) {
        int m = grid.length, n = grid[0].length, connected = 0;
        int[] cols = new int[n], last_server_col = new int[m];
        Arrays.fill(last_server_col, -1);

        for (int r = 0; r < m; r++) {
            int rows = 0;
            for (int c = 0; c < n; c++) {
                if (grid[r][c] == 1) {
                    rows++;
                    cols[c]++;
                    last_server_col[r] = c;
                }
            }

            if (rows != 1) {
                connected += rows;
                last_server_col[r] = -1;
            }
        }

        for (int r = 0; r < m; r++) {
            if (last_server_col[r] != -1 && cols[last_server_col[r]] > 1) {
                connected++;
            }
        }

        return connected;
    }
}